File: problem22/p22.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main()
{
    int fileSize, i = 0, j, sum = 0, N = 5163;
    char *content, *split, *names[5163], *temp;
    FILE *fp = fopen("C:/Users/ErenS/Desktop/names.txt", "rb");

    if (fp == NULL) {
        printf("Error while opening the file!\n");
        return 0;
    }

    fseek(fp, 0, SEEK_END);
    fileSize = ftell(fp);
    fseek(fp, 0, SEEK_SET);

    content = malloc(fileSize + 1);
    fread(content, 1, fileSize, fp);
    fclose(fp);

    split = strtok(content, ",");

    while (split != NULL) {
        names[i++] = split;
        split = strtok (NULL, ",");
    }

    for (i=0; i<N-1; i++) {
        for (j=0; j<N-1; j++) {
            if (strcmp(names[j], names[j+1]) > 0) {
                temp = names[j];
                names[j] = names[j+1];
                names[j+1] = temp;
            }
        }
    }

    for (i=0; i<5163; i++) {
        for (j=0; j<strlen(names[i]); j++) {
            if (names[i][j] != '"'){
                sum += ((int)names[i][j] - 64) * (i + 1);
            }
        }
    }

    printf("%d\n", sum);

    return 0;
}

File: problem22/p22.java
import java.io.*;
import java.util.*;

public class p22 {
	
	public static int namesScores(String fileName) {
		
		int i, j, sum = 0;
		String[] splitted;
		String name, content;
		List<String> names;
		File file = new File(fileName);
		byte[] data = new byte[(int) file.length()];
		
		try(FileInputStream fis = new FileInputStream(file)) {
			fis.read(data);
		} catch (FileNotFoundException ex) {
			System.out.println(ex);
		} catch (IOException ex) {
			System.out.println(ex);
		}
		
		try {
			content = new String(data, "UTF-8");
			splitted = content.split(",");
			names = Arrays.asList(splitted);
			Collections.sort(names);
			
			for(i=0; i<names.size(); i++) {
				name = names.get(i).replaceAll("\"", "");
				for(j=0; j<name.length(); j++) {
					sum += ((int)name.charAt(j) - 64) * (i + 1);
				}
			}
			
		} catch (UnsupportedEncodingException ex) {
			System.out.println(ex);
		}
		
		return sum;
	}
	
	public static void main(String[] args) {
		
		System.out.println(namesScores("C:\\Users\\ErenS\\Desktop\\names.txt"));
		
	}

}
